src/wallet/scriptpubkeyman.cpp

```diff
@@ -1560,12 +1560,14 @@ bool DescriptorScriptPubKeyMan::IsHDEnabled() const
 
 bool DescriptorScriptPubKeyMan::CanGetAddresses(bool internal) const
 {
-    return false;
+    LOCK(cs_desc_man);
+    return HavePrivateKeys() || descriptor.next_index < descriptor.range_end;
```

Comment:
It isn't. `CanGetAddresses` should return false for non-ranged descriptors because you cannot get new addresses from such a descriptor. `HavePrivateKeys` here is for when the cache runs on hardened derivation. If we have private keys, then we can continue to derive those hardened keys.

Reply:
> CanGetAddresses should return false for non-ranged descriptors because you cannot get new addresses from such a descriptor

Hm? If you import a non-ranged descriptor that includes a private key, `CanGetAddress` will return true here.

Reply:
> But we may still be unable to get addresses, these conditions are caught later.

from `GetNewDestination`. Might want to move that comment to where I'm asking and explain exactly what additional checks are required?

Reply:
I've tightened up the checks that `CanGetAddresses` does. It will do a `IsSingleType` and `IsRange` check now to disallow non-ranged descriptor and combo descriptors.